src/converse-register.js

```diff
@@ -227,18 +233,36 @@
                         $domain_input.addClass('error');
                         return;
                     }
-                    $form.find('input[type=submit]').hide()
-                        .after(tpl_registration_request({
-                            cancel: __('Cancel'),
-                            info_message: __('Requesting a registration form from the XMPP server')
-                        }));
-                    $form.find('button.button-cancel').on('click', this.cancelRegistration.bind(this));
+                    $form.find('input[type=submit]').hide();
+                    this.renderRegistrationRequest(__('Cancel'));
+                    this.fetchRegistrationForm(domain);
+                },
+
+                fetchRegistrationForm: function(domain_name) {
+                    /* This is called with a domain name based on which, it fetches a
+                     * registration form from the requested domain.
+                     *
+                     * Parameters:
+                     *      (Domain name) domain_name - XMPP server domain
+                     */ 
                     this.reset({
-                        domain: Strophe.getDomainFromJid(domain),
+                        domain: Strophe.getDomainFromJid(domain_name),
                         _registering: true
                     });
                     _converse.connection.connect(this.domain, "", this.onRegistering.bind(this));
-                    return false;
+                    return false; 
+                },
+
+                renderRegistrationRequest: function(cancel_label) {
+                    var form_help = document.querySelector('.form-help');
+                    $(form_help).after(tpl_registration_request({
+                        cancel: cancel_label,
+                        info_message: _converse.__('Requesting a registration form from the XMPP server')
+                    }));
+                    if (!_converse.registration_domain) {
+                        var cancel_button = document.querySelector('button.button-cancel');
+                        $(cancel_button).on('click', this.cancelRegistration.bind(this));
```

Comment:
I know you struggled with `insertAdjacentHTML` so it's ok to use `after` above until we remove jQuery completely from this plugin, but here you can already stop using jQuery.

Use `cancel_button.addEventListener('click', this.cancelRegistration.bind(this));`